common/src/stack/pylib/stack/probepal/common.py

```diff
@@ -0,0 +1,104 @@
+from abc import ABC, abstractmethod
+from operator import attrgetter
+import pathlib
+import importlib
+import pkgutil
+import inspect
+from dataclasses import dataclass
+
+class UnrecognizedPallet(Exception):
+	pass
+
+@dataclass(eq=True, frozen=True)
+class PalletInfo:
+	name: str
+	version: str
+	release: str
+	arch: str
+	distro_family: str
+
+	def is_complete(self):
+		return all((self.name, self.version, self.release, self.arch, self.distro_family))
+
+
+class Probe(ABC):
+	'''
+	Base probe class.  Subclasses must implement probe() and __init__()
+	Lower weight probes will be attempted first.
+
+	Subclassed `Probe.probe()` must return None or a PalletInfo() object
+
+	Probe instances are callables. `__call__(arg)` will pass through to `Probe.probe()`
+	'''
+
+	isProbe = True
+
+	@abstractmethod
+	def __init__(self, weight=90, desc=''):
```

Comment:
I would recommend requiring a description by not defaulting it to the empty string.